packages/eslint-plugin/src/rules/no-redundant-type-constituents.ts

```diff
@@ -323,8 +327,58 @@ export default createRule({
               }
             }
           }
-        }
 
+          typePartFlags.length >= 2
+            ? seenUnionTypes.set(typeNode, typePartFlags)
+            : null;
+        }
+        const checkIfUnionsAreAssignable = (): undefined => {
+          let typeFlagsOfUnions: ts.TypeFlags[] = [];
+          let result = true;
+          let primitiveUnit: number;
+
+          seenUnionTypes.forEach((value, key) => {
+            value.forEach(union => {
+              typeFlagsOfUnions.push(union.typeFlags);
+            });
+            for (const iterator of typeFlagsOfUnions) {
+              if (
+                seenPrimitiveTypes.has(
+                  literalToPrimitiveTypeFlags[
+                    iterator as keyof typeof literalToPrimitiveTypeFlags
+                  ],
+                )
+              ) {
+                result = true;
+                primitiveUnit =
+                  literalToPrimitiveTypeFlags[
+                    iterator as keyof typeof literalToPrimitiveTypeFlags
+                  ];
+              } else {
+                result = false;
+                break;
+              }
+            }
+            if (result) {
+              context.report({
+                data: {
+                  literal: value.map(name => name.typeName).join(' | '),
+                  primitive:
+                    primitiveTypeFlagNames[
+                      primitiveUnit as keyof typeof primitiveTypeFlagNames
+                    ],
+                },
+                messageId: 'primitiveOverridden',
+                node: key,
+              });
+            }
+            typeFlagsOfUnions = [];
```

Comment:
[Refactor] Unit tests still pass when this is removed. Any particular reason to keep it around?

```suggestion
```